### PR TITLE
feat: add retry logic for MetaPeerClient

### PR DESCRIPTION
**Pull request by Fengys123**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

This pr mainly change:  add retry logic for MetaPeerClient.

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

### CODE REVIEW

File: src/meta-srv/src/bootstrap.rs [hunk 1]
```diff
@@ -91,7 +91,14 @@ pub async fn make_meta_srv(opts: MetaSrvOptions) -> Result<MetaSrv> {
...
 
     let in_memory = Arc::new(MemStore::default()) as ResetableKvStoreRef;
-    let meta_peer_client = MetaPeerClient::new(in_memory.clone(), election.clone());
+
+    let mut builder = MetaPeerClientBuilder::default();
+    let meta_peer_client = builder
+        .election(election.clone())
+        .in_memory(in_memory.clone())
+        .build()
+        // Safety: all required fields set at initializatio
```
Comment (MichaelScofield) [inline src/meta-srv/src/bootstrap.rs:100]: typo: "initializatio"

File: src/meta-srv/src/cluster.rs [hunk 2]
```diff
@@ -13,52 +13,93 @@
...
 
-#[derive(Clone)]
+#[derive(Builder, Clone)]
 pub struct MetaPeerClient {
     election: Option<ElectionRef>,
     in_memory: ResetableKvStoreRef,
+    #[builder(default = "ChannelManager::default()")]
     channel_manager: ChannelManager,
+    #[builder(default = "3")]
+    retry_num: usize,
```
Comment (MichaelScofield) [inline src/meta-srv/src/cluster.rs:41]:
```suggestion
    max_retry_count: usize,
```

File: src/meta-srv/src/cluster.rs [hunk 3]
```diff
@@ -13,52 +13,93 @@
...
+    #[builder(default = "1000")]
+    interval_ms: u64,
```
Comment (MichaelScofield) [inline src/meta-srv/src/cluster.rs:43]:
```suggestion
    retry_interval_ms: u64,
```

File: src/meta-srv/src/service/cluster.rs [hunk 4]
```diff
@@ -31,6 +32,8 @@ impl cluster_server::Cluster for MetaSrv {
                 header: Some(is_not_leader),
                 ..Default::default()
             };
+
+            warn!("The current meta is not leader, but a batch_get request have reached the meta.");
```
Comment (MichaelScofield) [inline src/meta-srv/src/service/cluster.rs:36]: Better to provide more detailed information like some context or request body if you decide to log them. Same to line 60.

File: src/meta-srv/src/bootstrap.rs (hunk 1 above)
Comment (Fengys123) [reply src/meta-srv/src/bootstrap.rs:100]: Good catch!

**Review by MichaelScofield — APPROVED** [review]

File: src/meta-srv/src/bootstrap.rs [hunk 5]
```diff
@@ -91,7 +91,14 @@ pub async fn make_meta_srv(opts: MetaSrvOptions) -> Result<MetaSrv> {
     };
 
     let in_memory = Arc::new(MemStore::default()) as ResetableKvStoreRef;
-    let meta_peer_client = MetaPeerClient::new(in_memory.clone(), election.clone());
+
+    let mut builder = MetaPeerClientBuilder::default();
+    let meta_peer_client = builder
```
Comment (shuiyisong) [inline src/meta-srv/src/bootstrap.rs:96]:
```suggestion
    let meta_peer_client = MetaPeerClientBuilder::default()
```

**Comment by Fengys123** [comment]
PTAL @shuiyisong

**Review by shuiyisong — APPROVED** [review]

**Pull request merged**